src/smtsolvers/CoreSMTSolver.cc

```diff
@@ -2146,12 +2146,10 @@ bool CoreSMTSolver::createSplit_scatter(bool last)
         // space
         constraints_negated.push(~l);
     }
-    for (int i = 0; i < split_assumptions.size()-1; i++)
-    {
+    for (const auto & split_assumption : split_assumptions) {
```

Comment:
The previous version was ignoring the last element of `split_assumptions`, the new version includes it. Which one is correct?

Reply:
Right, this is a bug as well. It should ignore the last element.